tests/console/commands/test_run.py

```diff
@@ -146,3 +146,40 @@ def test_run_script_exit_code(
     )
     assert tester.execute("exit-code") == 42
     assert tester.execute("return-code") == 42
+
+
+@pytest.mark.parametrize(
+    "installed_script", [False, True], ids=["not installed", "installed"]
+)
+def test_run_script_sys_argv0(
+    installed_script: bool,
+    poetry_with_scripts: Poetry,
+    command_tester_factory: CommandTesterFactory,
+    tmp_venv: VirtualEnv,
+    mocker: MockerFixture,
+) -> None:
+    """
+    If RunCommand calls an installed script defined in pyproject.toml,
+    sys.argv[0] must be set to the full path of the script.
+    """
+    mocker.patch("poetry.utils.env.EnvManager.get", return_value=tmp_venv)
+    mocker.patch(
+        "os.execvpe",
+        lambda file, args, env: subprocess.call([file] + args[1:], env=env),
+    )
+
+    install_tester = command_tester_factory(
+        "install",
+        poetry=poetry_with_scripts,
+        environment=tmp_venv,
+    )
+    assert install_tester.execute() == 0
+    if not installed_script:
+        for path in tmp_venv.script_dirs[0].glob("check-argv0*"):
+            path.unlink()
```

Comment:
Here the test would be more clear if I don't install at all when `installed_script == True`.

I actually tried to refactor this way, but for some reason, the test for "not installed" fails...

Here what I tried
```py
    if installed_script:
        install_tester = command_tester_factory(
            "install",
            poetry=poetry_with_scripts,
            environment=tmp_venv,
        )
        assert install_tester.execute() == 0
```

Reply:
Never mind this one. For some reason, I believed that I could use `poetry run` before issue a `poetry install`.

Reply:
Yep, not installing at all was my first approach. However, it seems the root package has to be installed. Only the script itself needn't to be installed.